unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BuilderInWorldEntityHandler.cs

```diff
@@ -62,16 +63,20 @@ private void Start()
         showAllEntitiesAction.OnTriggered += showAllEntitiesDelegate;
     }
 
-    public void Init()
+    public override void Init()
     {
-        HUDController.i.builderInWorldMainHud.OnEntityDelete += DeleteSingleEntity;
-        HUDController.i.builderInWorldMainHud.OnDuplicateSelectedAction += DuplicateSelectedEntitiesInput;
-        HUDController.i.builderInWorldMainHud.OnDeleteSelectedAction += DeleteSelectedEntitiesInput;
-        HUDController.i.builderInWorldMainHud.OnEntityClick += ChangeEntitySelectionFromList;
-        HUDController.i.builderInWorldMainHud.OnEntityLock += ChangeEntityLockStatus;
-        HUDController.i.builderInWorldMainHud.OnEntityChangeVisibility += ChangeEntityVisibilityStatus;
-        HUDController.i.builderInWorldMainHud.OnEntityRename += SetEntityName;
-        HUDController.i.builderInWorldMainHud.OnEntitySmartItemComponentUpdate += UpdateSmartItemComponentInKernel;
+        base.Init();
+        if (HUDController.i.builderInWorldMainHud != null)
+        {
+            HUDController.i.builderInWorldMainHud.OnEntityDelete += DeleteSingleEntity;
```

Comment:
Maybe we can store the `HUDController.i.builderInWorldMainHud` in a local variable to increase legibility.

Reply:
Done, however, we need to take into account that if we recreate the UI somehow we lose the reference

Reply:
We could use a field for that:
`private BuilderInWorldMainHUD biwMainHUD => HUDController.i.builderInWorldMainHud`

Besides increasing readability, it's a good practice to follow to ease the process of searching for references.